### PR TITLE
Kadir Bozkurt Javascript Homework Week 2

### PR DESCRIPTION
**Pull request by kadirbozkurt05**

### CODE REVIEW

File: 1-JavaScript/Week2/homework/ex4-shoppingCart.js [hunk 1]
```diff
@@ -1,67 +1,56 @@
...
------------------------------------------------------------------------------*/
 const shoppingCart = ['bananas', 'milk'];
 
-// ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(item) {
+  if(item){
+    shoppingCart.push(item);
+  if(shoppingCart.length===4){
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex4-shoppingCart.js:8]: Very good. 
💡 Can also be `shoppingCart.length > 3`

File: 1-JavaScript/Week2/homework/ex4-shoppingCart.js [hunk 2]
```diff
@@ -1,67 +1,56 @@
...
+    shoppingCart.shift();
+  }
+}
+  return `You bought ${shoppingCart.join(', ')}!`;
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex4-shoppingCart.js:12]: Very nice that you found the `join` function - it is very useful!

File: 1-JavaScript/Week2/homework/ex5-shoppingCartPure.js [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
+function addToShoppingCart(arr,item) {
+  const shoppingCart =[...arr];
+
+  if(item){
+    shoppingCart.push(item);
+    if(shoppingCart.length===4){
+      shoppingCart.shift();
+    }
+  }
+  return shoppingCart
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex5-shoppingCartPure.js:11]: Good job! Nice and clean

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 4]
```diff
@@ -1,42 +1,34 @@
...
 function test1() {
   console.log('\nTest 1: calculateTotalPrice should take one parameter');
-  // TODO replace this comment with your code
+  console.assert(calculateTotalPrice.length === 1);
 }
 
 function test2() {
   console.log('\nTest 2: return correct output when passed cartForParty');
-  // TODO replace this comment with your code
+  console.assert(calculateTotalPrice(cartForParty).includes(15.48));
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:31]: Nice tests!

File: 1-JavaScript/Week2/homework/ex6-totalCost.js [hunk 5]
```diff
@@ -1,42 +1,34 @@
...
+  console.log(cartKeys);
 
-3. Complete the unit test functions and verify that all is working as expected.
------------------------------------------------------------------------------*/
-const cartForParty = {
-  // TODO complete this object
-};
+  for (const key of cartKeys) {
+    total += cart[key];
+  }
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex6-totalCost.js:19]: Well done looping over the keys of the array.

💡  By the way, you can also use the `for...in` loop on objects:
` for (const key in cart) { ... }`

File: 1-JavaScript/Week2/homework/ex6-totalCost.js (hunk 4 above)
Comment (stasel) [reply 1-JavaScript/Week2/homework/ex6-totalCost.js:31]: It's also ok to have the full string here in the assert:
`console.assert(calculateTotalPrice(cartForParty) === "Total: €15.48");`

File: 1-JavaScript/Week2/homework/ex7-mindPrivacy.js [hunk 6]
```diff
@@ -29,12 +17,17 @@ const employeeRecords = [
...
-  // TODO complete this function
+function filterPrivateData(recordsArray) {
+  const filteredRecords = [];
+
+  for(const employee of recordsArray){
+    const {name,occupation,email} = employee;
+    filteredRecords.push({name,occupation,email});
+  }
+  return filteredRecords;
+
```
Comment (stasel) [inline 1-JavaScript/Week2/homework/ex7-mindPrivacy.js:28]: Perfect! Clean and readable function 👍

**Review by stasel — APPROVED** [review]
Dear Kadir,
You did a wonderful job in this week's homework.
Your code is clear, readable and works as expected! You seem to understand how to write good JS code!

✅

**Comment by kadirbozkurt05** [comment]
Thanks for your advises. I appreciate your time 🙏

**Pull request closed without merging**